spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/procedures/RemoveSnapshotsProcedure.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.spark.procedures;
+
+import org.apache.iceberg.ExpireSnapshots;
+import org.apache.iceberg.spark.procedures.SparkProcedures.ProcedureBuilder;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.connector.catalog.Identifier;
+import org.apache.spark.sql.connector.catalog.TableCatalog;
+import org.apache.spark.sql.connector.iceberg.catalog.ProcedureParameter;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.Metadata;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A procedure that removes corrupt snapshots whose ManifestList files are missing. To expire
+ * healthy snapshots, use ExpireSnapshotsProcedure.
+ */
+public class RemoveSnapshotsProcedure extends BaseProcedure {
```

Comment:
I'm a bit sceptical that introducing a new procedure is the best approach here. I think it will cause more confusion for users as they won't know when to use which. Wouldn't it be better to add a flag to the existing procedure and properly document it?

Reply:
I attempted with #9584. Then I realized it can remove healthy snapshots unless I do some hacks like `older_than => epoch timestamp` to keep all other snapshots. 

One thing I find counterintuitive with `expire_snapshots` is with `snapshot_ids` specified, it will still remove other expired snapshots. This behavior doesn't work with what I want to achieve.